src/smart-components/overview/enable-workspaces-alert.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { UsersIcon } from '@patternfly/react-icons';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Checkbox,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Switch,
+  Text,
+  TextVariants,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
+import { useIntl } from 'react-intl';
+import Messages from '../../Messages';
+import './enable-workspaces-alert.scss';
+
+const EnableWorkspacesAlert: React.FunctionComponent = () => {
+  const [isToggled, setIsToggled] = React.useState<boolean>(false);
+  const [checked, setChecked] = React.useState<boolean>(false);
+  const [isModalOpen, setIsModalOpen] = React.useState<boolean>(false);
+  const [isConfirmed, setIsConfirmed] = React.useState<boolean>(false);
+  const intl = useIntl();
+
+  const handleModalToggle = () => {
+    setIsToggled(false);
+    setChecked(false);
+    setIsConfirmed(false);
+    setIsModalOpen((prevIsModalOpen) => !prevIsModalOpen);
+    document.getElementById('enable-workspaces-switch')?.click();
+  };
```

Comment:
```suggestion
const onClose = () => {
    setChecked(false);
    setIsModalOpen(false);
  };
```

Reply:
we should not use `document.getElementById`, but at least refs if it is really needed, but in this case, `defaultChecked` prop blocked the updates. Now you can just change the value and get rid of the click

Reply:
I knew I was missing something silly. Thank you, Filip!